src/main/java/org/jenkinsci/plugins/docker/commons/tools/DockerToolInstaller.java

```diff
@@ -142,6 +146,27 @@ public FilePath performInstallation(ToolInstallation toolInstallation, @Nonnull
         return install;
     }
 
+    static URL getDockerImageUrl(String os, String version) throws MalformedURLException {
+        if (parseVersion(version).isNewerThan(parseVersion("17.05.0-ce")))
+            return new URL("https://download.docker.com/" + os + "/docker-" + version);
+
+        String osName="";
+        if (os.startsWith("linux")) osName = "Linux";
+        if (os.startsWith("win")) osName = "Windows";
+        if (os.startsWith("mac")) osName = "Darwin";
+        return new URL("https://get.docker.com/builds/" + osName + os.substring(os.lastIndexOf("/")) + "/docker-" + version);
+    }
+
+    private static VersionNumber parseVersion(String version) {
+        // any version that sorts before 17.05.0-ce
+        if (version.equals("latest")) return new VersionNumber("0");
```

Comment:
Would `latest` not be “newer than” `17.05.0-ce`? Your test does request this go to `get.docker.com`. Yet https://get.docker.com/builds/Linux/x86_64/docker-latest is a 403. OTOH https://download.docker.com/linux/static/stable/x86_64/docker-latest is a 404. It seems `latest` just does not work any more—the index of https://download.docker.com/linux/static/ lists `edge`, `stable`, and `test`, but the file names in each specify a version. However I will check the test behavior and see what is going on.

Reply:
Ah, https://get.docker.com/builds/Linux/x86_64/docker-latest.tgz is a 200.

Reply:
…and it is `17.05.0-ce`. So they are still publishing to `get.docker.com`, just not the actual “latest” (which seems to be `17.12.0-ce` currently). Weird.